src/main/java/org/opensearch/timeseries/indices/IndexManagement.java

```diff
@@ -283,7 +289,7 @@ protected void choosePrimaryShards(CreateIndexRequest request, boolean hiddenInd
             );
     }
 
-    protected void deleteOldHistoryIndices(String indexPattern, TimeValue historyRetentionPeriod) {
+    protected void deleteOldHistoryIndices(String indexPattern, TimeValue historyRetentionPeriod, Integer customResultIndexTtl) {
```

Comment:
This is running for default result indices. But we want to check each custom index's custom ttl and do it one by one. Also, since there can be a lot of custom indices and the method is async, you may want to run the delete in batches. Finish batch of delete before the other custom indices.

Reply:
Also, can we still keep current method signature?

void deleteOldHistoryIndices(String indexPattern, TimeValue historyRetentionPeriod)

If it is default index, you called it the old way. If it is custom index, you translate customResultIndexTtl of type integer to historyRetentionPeriod of type TimeValue.

Reply:
kept current method signature. 

Not sure what you mean by run the delete in batches. Right now in deleteOldHistoryIndices, it seems like we already collecting candidates indices for deletion into a list, then perform deletion. Let's sync offline about this